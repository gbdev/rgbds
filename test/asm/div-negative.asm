MACRO test
	def num = \1
	def den = \2
	def quo = num / den
	def rem = num % den
	def rev = quo * den + rem
	assert num == rev
	println "{ 11d:num} (${08x:num}) / { 11d:den} (${08x:den}) = { 11d:quo} (${08x:quo}) R { 11d:rem} (${08x:rem})"
ENDM

test $8000_0000, $8000_0000
test $8000_0000, $c000_0000
test $8000_0000, $f000_0000
test $8000_0000, $f800_0000
test $8000_0000, $ff00_0000

test $c000_0000, $8000_0000
test $c000_0000, $c000_0000
test $c000_0000, $f000_0000
test $c000_0000, $f800_0000
test $c000_0000, $ff00_0000

test $f800_0000, $8000_0000
test $f800_0000, $c000_0000
test $f800_0000, $f000_0000
test $f800_0000, $f800_0000
test $f800_0000, $ff00_0000

test $0000_0000, $8000_0000
test $0000_0000, $c000_0000
test $0000_0000, $f000_0000
test $0000_0000, $f800_0000
test $0000_0000, $ff00_0000

test $0080_0000, $ff80_0000

test $0300_0000, $fd00_0000
test $0300_0000, $ff00_0000
test $0300_0000, $ff80_0000

test $6000_0000, $fd00_0000
test $6000_0000, $ff00_0000
test $6000_0000, $ff80_0000

test $7f00_0000, $fd00_0000
test $7f00_0000, $ff00_0000
test $7f00_0000, $ff80_0000
