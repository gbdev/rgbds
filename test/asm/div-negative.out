-2147483648 ($80000000) / -2147483648 ($80000000) =           1 ($00000001) R           0 ($00000000)
-2147483648 ($80000000) / -1073741824 ($c0000000) =           2 ($00000002) R           0 ($00000000)
-2147483648 ($80000000) /  -268435456 ($f0000000) =           8 ($00000008) R           0 ($00000000)
-2147483648 ($80000000) /  -134217728 ($f8000000) =          16 ($00000010) R           0 ($00000000)
-2147483648 ($80000000) /   -16777216 ($ff000000) =         128 ($00000080) R           0 ($00000000)
-1073741824 ($c0000000) / -2147483648 ($80000000) =           0 ($00000000) R -1073741824 ($c0000000)
-1073741824 ($c0000000) / -1073741824 ($c0000000) =           1 ($00000001) R           0 ($00000000)
-1073741824 ($c0000000) /  -268435456 ($f0000000) =           4 ($00000004) R           0 ($00000000)
-1073741824 ($c0000000) /  -134217728 ($f8000000) =           8 ($00000008) R           0 ($00000000)
-1073741824 ($c0000000) /   -16777216 ($ff000000) =          64 ($00000040) R           0 ($00000000)
 -134217728 ($f8000000) / -2147483648 ($80000000) =           0 ($00000000) R  -134217728 ($f8000000)
 -134217728 ($f8000000) / -1073741824 ($c0000000) =           0 ($00000000) R  -134217728 ($f8000000)
 -134217728 ($f8000000) /  -268435456 ($f0000000) =           0 ($00000000) R  -134217728 ($f8000000)
 -134217728 ($f8000000) /  -134217728 ($f8000000) =           1 ($00000001) R           0 ($00000000)
 -134217728 ($f8000000) /   -16777216 ($ff000000) =           8 ($00000008) R           0 ($00000000)
          0 ($00000000) / -2147483648 ($80000000) =           0 ($00000000) R           0 ($00000000)
          0 ($00000000) / -1073741824 ($c0000000) =           0 ($00000000) R           0 ($00000000)
          0 ($00000000) /  -268435456 ($f0000000) =           0 ($00000000) R           0 ($00000000)
          0 ($00000000) /  -134217728 ($f8000000) =           0 ($00000000) R           0 ($00000000)
          0 ($00000000) /   -16777216 ($ff000000) =           0 ($00000000) R           0 ($00000000)
    8388608 ($00800000) /    -8388608 ($ff800000) =          -1 ($ffffffff) R           0 ($00000000)
   50331648 ($03000000) /   -50331648 ($fd000000) =          -1 ($ffffffff) R           0 ($00000000)
   50331648 ($03000000) /   -16777216 ($ff000000) =          -3 ($fffffffd) R           0 ($00000000)
   50331648 ($03000000) /    -8388608 ($ff800000) =          -6 ($fffffffa) R           0 ($00000000)
 1610612736 ($60000000) /   -50331648 ($fd000000) =         -32 ($ffffffe0) R           0 ($00000000)
 1610612736 ($60000000) /   -16777216 ($ff000000) =         -96 ($ffffffa0) R           0 ($00000000)
 1610612736 ($60000000) /    -8388608 ($ff800000) =        -192 ($ffffff40) R           0 ($00000000)
 2130706432 ($7f000000) /   -50331648 ($fd000000) =         -43 ($ffffffd5) R   -33554432 ($fe000000)
 2130706432 ($7f000000) /   -16777216 ($ff000000) =        -127 ($ffffff81) R           0 ($00000000)
 2130706432 ($7f000000) /    -8388608 ($ff800000) =        -254 ($ffffff02) R           0 ($00000000)
